Sources/Fullscreen/SearchResultMapView/SearchResultMapView.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import MapKit
+
+public protocol SearchResultMapViewDelegate: AnyObject {
+    func searchResultMapViewDidSelectChangeMapTypeButton(_ view: SearchResultMapView, mapSettingsButtonView: MapSettingsButton)
+    func searchResultMapViewDidSelectCenterMapButton(_ view: SearchResultMapView)
+    func searchResultMapViewDidSelectAnnotationView(_ view: SearchResultMapView, annotationView: MKAnnotationView)
+    func searchResultMapViewRegionWillChangeDueToUserInteraction(_ view: SearchResultMapView)
+    func searchResultMapViewRegionDidChange(_ view: SearchResultMapView)
+    func searchResultMapViewDidUpdateUserLocation(_ view: SearchResultMapView, userLocation: MKUserLocation)
+}
+
+public final class SearchResultMapView: UIView {
+
+    private var didSetupView = false
+
+    public weak var delegate: SearchResultMapViewDelegate?
+
+    public var zoomLevel: Double {
+        return mapView.zoomLevel
+    }
+
+    public var centerCoordinate: CLLocationCoordinate2D {
+        return mapView.centerCoordinate
+    }
+
+    public var visibleMapRect: MKMapRect {
+        return mapView.visibleMapRect
+    }
+
+    private lazy var mapView: MKMapView = {
+        let view = MKMapView()
+        view.delegate = self
+        view.isRotateEnabled = false
+        view.isPitchEnabled = false
+        return view
+    }()
+
+    private lazy var mapSettingsButton: MapSettingsButton = {
+        let mapSettingsButton = MapSettingsButton(withAutoLayout: true)
+        mapSettingsButton.delegate = self
+        return mapSettingsButton
+    }()
+
+    // MARK: - Public
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        if !didSetupView {
```

Comment:
Why doing the setup in a custom layoutSubviews instead of in an init? Doing it in the init will remove the need to have `didSetupView` or am I missing something? 🤔 :)

Reply:
You're right! This is probably the consequence of me looking at a similar view and trying to adapt the same patterns. Will change it to an `init` ☺️